client/files.go

```diff
@@ -86,7 +86,7 @@ type errorResult struct {
 func (client *Client) MakeDir(opts *MakeDirOptions) error {
 	var permissions string
 	if opts.Permissions != 0 {
-		permissions = fmt.Sprintf("%03o", opts.Permissions)
+		permissions = fmt.Sprintf("%04o", opts.Permissions)
```

Comment:
We should still make this `%03o` so the client works with older Pebble servers (which checks for exactly 3). That will make it 3 digits, unless you pass in a higher number like 0o1755, in which case it'll be 4 digits (which won't work on older clients anyway).